src/Operations/Bitwise.py

```diff
@@ -1,74 +1,177 @@
 from Operations.Binary import Binary
 from Operations.Unary import Unary
+from Operations.Arithmetics import Sdiv
 from Imports import UINT256_MODULE
+from utils import bit_not, is_bit_set
 
 
 class And(Binary):
-    def bind_to_res(self, op1, op2, res):
-        return f"let (local {res} : Uint256) = uint256_and({op1}, {op2})"
+    def bind_to_res(self, op1, op2, res, evaluatable):
+        if evaluatable:
+            evaluated = op1 & op2
+            res_high, res_low = divmod(evaluated, 2 ** 128)
+            return (
+                f"local {res} : Uint256 = Uint256({res_low}, {res_high})",
+                True,
+                evaluated,
+            )
+        else:
+            return (
+                f"let (local {res} : Uint256) = uint256_and({op1}, {op2})",
+                False,
+                0,
+            )
```

Comment:
It seems like almost all of the changes related to operation classes follow the same pattern. The only thing that differs is the evaluation function in python and the evaluation function in cairo. And there is a lot of such classes. Perhaps it would make sense to extract the common logic in the parent class?